File: Software Development Module/Week12/PhoneNumberApp.java
/*
PhoneNumberApp.java
Billy Cussen
23/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class PhoneNumberApp{

    public static void main(String[]args){

        //Variables
        String input = new String();
        String output;
        String retry;
        boolean inputCheck;

        //Object
        Scanner keyboard;
        PhoneNumber myPhoneNumber;

        do{

            do{
                keyboard = new Scanner(System.in);
                inputCheck = true;

                try{
                    System.out.print("Please enter your phone number: ");
                    input = keyboard.nextLine();

                    if(input.length()!=11){
                        throw new InputMismatchException("Your Phone number is not the correct length");
                    }

                    int checkDash = 0;

                    for(int k = 0; k < input.length(); k++){
                        if(input.charAt(k)=='-'){
                            checkDash++;
                        }
                    }

                    if(checkDash==1){
                        checkDash = input.indexOf("-");
                    }else{
                        throw new InputMismatchException("You need to have one Dash between the Prefix and Main Number: 000-0000000");
                    }

                    if(input.charAt(checkDash)!='-'){
                        throw new InputMismatchException("You need to have one Dash between the Prefix and Main Number: 000-0000000");
                    }

                    for(int i = 0; i < checkDash;i++){
                        if(!(Character.isDigit(input.charAt(i)))){
                            throw new InputMismatchException("Your phone number needs to be in the following format: 000-0000000");
                        }
                    }

                    for(int j=checkDash+1;j<input.length();j++){
                        if(!(Character.isDigit(input.charAt(j)))){
                            throw new InputMismatchException("Your phone number needs to be in the following format: 000-0000000");
                        }
                    }

                }catch(InputMismatchException e){
                    System.out.println(e.getMessage());
                    inputCheck = false;
                }finally{
                    keyboard = null;
                }

            }while(!inputCheck);

            myPhoneNumber = new PhoneNumber(input);
            output = myPhoneNumber.computeAndGetResult();
            myPhoneNumber = null;
            System.out.println(output);

            do{
                keyboard = new Scanner(System.in);
                retry = new String();

                try{
                    System.out.print("Retry (Y/N): ");
                    retry = keyboard.nextLine();

                    if(!(retry.equalsIgnoreCase("Y")||retry.equalsIgnoreCase("N"))){
                        throw new InputMismatchException();
                    }

                }catch(InputMismatchException e){
                    System.out.println("You need to enter Y or N. ");
                }finally{
                    keyboard = null;
                }   
            }while(!(retry.equalsIgnoreCase("Y")||retry.equalsIgnoreCase("N")));

        }while(retry.equalsIgnoreCase("Y"));
    }

}

File: Python Course/Software Development Module - in Python/Week12/PhoneNumberApp.py
"""
PhoneNumberApp.py
Billy Cussen
15/10/2020
"""

#Objects
import gc
import PhoneNumber
from PhoneNumber import myPhoneNumber

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = False

    #Input - Phone Number
    while inputCheck == False:
        try:
            phoneNumber = str(input("Please enter a Phone Number: "))
        except Exception as e:
            print("You need to enter a valid phone number")
        else:
            if(phoneNumber.find("-")!=3):
                print("This phone number is invalid, it needs a \"-\" after the prefix (e.g. 083-1234556)")
            elif(len(phoneNumber)!=11):
                print("Your phone number needs to have a length of 11 (e.g. 083-4444444)")
            else:
                inputCheck = True

    #Process and Output
    myPhoneNumberApp = myPhoneNumber(phoneNumber)
    myPhoneNumberApp.computePhoneNumber()
    myPhoneNumberApp = None
    gc.collect()

    #Input - Repeat?
    repeat = ""
    if repeat != "y" and repeat !="n":
        repeat = str(input("Please enter \"y\" to retry, \"n\" to Exit: "))
        if repeat != "y" and repeat != "n":
            print("Please enter \"y\" or \"n\"")
        

print("Thanks for using my Phone Number App!")